pkg/typesystem/typesystem.go

```diff
@@ -1066,11 +1078,35 @@ func (t *TypeSystem) validateTypeRestrictions(objectType string, relationName st
 				}
 			}
 		}
+
+		if related.Condition != "" {
+			// Validate the conditions referenced by the relations are included in the model.
+			if _, ok := t.conditions[related.Condition]; !ok {
+				return ErrConditionUndefined
+			}
+		}
 	}
 
 	return nil
 }
 
+// validateConditions validates the conditions provided in the model.
+//  1. Each condition should define a well-formed expression,
+//  2. Its parameters should be well-defined
+//  3. The expression and parameters should be compilable.
```

Comment:
1 and 3 seem to say the same thing? or is 1 that the expression must return a boolean?

Also where are we doing 2 in the code?

Reply:
at this point its mostly in `(*EvaluableCondition).Compile` which already has a comment that i think covers it, will remove this bit.